### PR TITLE
Added support for `VectorIntoWasmAbi` and `VectorFromWasmAbi`

### PR DESCRIPTION
**Pull request by eneoli**
In [this pull request](https://github.com/rustwasm/wasm-bindgen/pull/3554) (see also: [this related issue](https://github.com/rustwasm/wasm-bindgen/issues/111)), the capability to return a ` Vec<SomeType>` was introduced to wasm-bindgen, where `SomeType` has the `#[wasm_bindgen]` attribute.

This PR further extends this capability to support returning `Vec<SomeType>` where `SomeType` derives Tsify. This is done by implementing `VectorIntoWasmAbi` and `VectorFromWasmAbi` introduced in the linked PR.

~~Two new attributes were added to control whether the above traits should be implemented.~~

Example:

```rust

#[derive(Serialize, Deserialize, Tsify)]
#[tsify(into_wasm_abi, from_wasm_abi)]
#[serde(tag = "kind", content = "value")]
pub enum Pet {
    Cat(String),
    Dog(String),
}

#[wasm_bindgen]
pub fn return_pets() -> Vec<Pet> {
    vec![Pet::Cat("Pumpkin"), Pet::Dog("Cookie")]
}

#[wasm_bindgen]
pub fn read_pets(pets: Vec<Pet>) {
    // ...
}
```

### CODE REVIEW

**Comment by eneoli** [comment]
@siefkenj  Sorry for the noise. I had an old clippy installed that did not detect the dead code. This also probably explains why the CI did not nag when this code was added first. (The dead code is unrelated to the PR.)

**Comment by siefkenj** [comment]
Thank you for the contribution! I will review it as soon as I can :-)

**Comment by siefkenj** [comment]
Is there a reason to put these behind a flag? Would it cause problems if `into_wasm_abi` implied `vector_into_wasm_abi`? (I would think that would be the default expected behavior)

Could you please add some tests to this PR?

File: tsify-next-macros/src/attrs.rs [hunk 1]
```diff
@@ -14,8 +12,6 @@ pub struct TsifyContainerAttrs {
     pub namespace: bool,
     /// Information about how the type should be serialized.
     pub ty_config: TypeGenerationConfig,
-    /// Comments associated with the type. These will be written out to the generated Typescript.
-    pub comments: Vec<String>,
```
Comment (siefkenj) [inline tsify-next-macros/src/attrs.rs:18]: Were these comments not being used? Why was this field removed?

File: tests-e2e/test1/entry_point.rs [hunk 2]
```diff
@@ -9,7 +9,22 @@ pub struct Point {
...
+pub fn consume_vector(points: Vec<Point>) {}
+
+#[wasm_bindgen]
+pub fn vector_into_js() -> Vec<Point> {
+    vec![
+        Point { x: 1, y: 6 },
+        Point { x: 2, y: 5 },
+        Point { x: 3, y: 4 },
+    ]
+}
```
Comment (siefkenj) [inline tests-e2e/test1/entry_point.rs]: This test looks great, but could you please create a new test folder so the previous test remain unchanged?

File: tsify-next-macros/src/attrs.rs (hunk 1 above)
Comment (eneoli) [reply tsify-next-macros/src/attrs.rs:18]: Correct, the static code analysis (clippy) reported that this field was never read. That was also one of the reasons the CI failed.

File: tests-e2e/test1/entry_point.rs (hunk 2 above)
Comment (eneoli) [reply tests-e2e/test1/entry_point.rs]: Done

**Comment by siefkenj** [comment]
Thanks for your effort!

**Comment by eneoli** [comment]
Thank you for merging my pull request!

Could you please also publish a new release that includes these changes?

**Comment by siefkenj** [comment]
Done :-)

**Comment by Pantamis** [comment]
Thank you so much ! 

This is really great stuff ! I was just missing this XD

**Pull request merged**